core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +560,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: ReplayEventsFilter = SELECT_ALL_FILTER): Option[FsApplicationAttemptInfo] = {
```

Comment:
I think I agree, but then, the other call site actually uses the return value of this method, but, also supplies a listener/bus to collect other info, and then also parses that. It seems like this method could simply do all that work for both callers and not even need an external bus?